File: detekt/wrapper/src/test/java/io/buildfoundation/bazel/detekt/execute/WorkerExecutableTests.java
package io.buildfoundation.bazel.detekt.execute;

import io.buildfoundation.bazel.detekt.value.WorkRequest;
import io.buildfoundation.bazel.detekt.value.WorkResponse;
import org.junit.Test;

import java.util.Arrays;

import static org.junit.Assert.assertEquals;

public class WorkerExecutableTests {

    @Test
    public void successResponseOnExecutionSuccess() {
        check(new ExecutableResult.Success());
    }

    @Test
    public void failureResponseOnExecutionFailure() {
        check(new ExecutableResult.Failure("fake message"));
    }

    private void check(ExecutableResult executableResult) {
        WorkerExecutable workerExecutable = new WorkerExecutable.Impl(new TestExecutable(executableResult));

        WorkRequest workRequest = new WorkRequest();
        workRequest.requestId = 42;
        workRequest.arguments = Arrays.asList("fake", "fake", "fake");

        WorkResponse workResponseActual = workerExecutable.execute(workRequest);

        WorkResponse workResponseExpected = new WorkResponse();
        workResponseExpected.requestId = workRequest.requestId;
        workResponseExpected.output = executableResult.output();
        workResponseExpected.exitCode = executableResult.statusCode();

        assertEquals(workResponseExpected, workResponseActual);
    }
}

File: detekt/wrapper/src/test/kotlin/io/buildfoundation/bazel/detekt/execute/WorkerExecutableTests.kt
package io.buildfoundation.bazel.detekt.execute

import bazel.worker.WorkerProtocol
import org.junit.Assert.assertEquals
import org.junit.Test

class WorkerExecutableTests {

    @Test
    fun `it returns successful response on successful execution`() {
        val request = WorkerProtocol.WorkRequest.newBuilder()
                .setRequestId(42)
                .build()

        val actualResponse = WorkerExecutable.Impl(TestExecutable(Result.Success)).execute(request)

        val expectedResponse = WorkerProtocol.WorkResponse.newBuilder()
                .setRequestId(request.requestId)
                .setOutput("")
                .setExitCode(0)
                .build()

        assertEquals(expectedResponse, actualResponse)
    }

    @Test
    fun `it returns failure response on failed execution`() {
        val request = WorkerProtocol.WorkRequest.newBuilder()
                .setRequestId(42)
                .build()

        val result = Result.Failure("result description")

        val actualResponse = WorkerExecutable.Impl(TestExecutable(result)).execute(request)

        val expectedResponse = WorkerProtocol.WorkResponse.newBuilder()
                .setRequestId(request.requestId)
                .setOutput(result.description)
                .setExitCode(1)
                .build()

        assertEquals(expectedResponse, actualResponse)
    }
}